src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs

```diff
@@ -55,54 +55,32 @@ public static int Start(string consoleFilePath, [MarshalAs(UnmanagedType.LPArray
             Thread.CurrentThread.CurrentUICulture = NativeCultureResolver.UICulture;
             Thread.CurrentThread.CurrentCulture = NativeCultureResolver.Culture;
 
-            RunspaceConfigForSingleShell configuration = null;
-            PSConsoleLoadException warning = null;
-            //      PSSnapInException will cause the control to return back to the native code
-            //      and stuff the EXCEPINFO field with the message of the exception.
-            //      The native code will print this out and exit the process.
-            if (string.IsNullOrEmpty(consoleFilePath))
-            {
 #if DEBUG
-// Special switches for debug mode to allow self-hosting on InitialSessionState instead
-// of runspace configuration...
-                    if (args.Length > 0 && !String.IsNullOrEmpty(args[0]) && args[0].Equals("-iss", StringComparison.OrdinalIgnoreCase))
-                    {
-                        ConsoleHost.DefaultInitialSessionState = InitialSessionState.CreateDefault2();
-                        configuration = null;
-                    }
-                    else if (args.Length > 0 && !String.IsNullOrEmpty(args[0]) && args[0].Equals("-isswait", StringComparison.OrdinalIgnoreCase))
-                    {
-                        Console.WriteLine("Attach the debugger and hit enter to continue:");
-                        Console.ReadLine();
-                        ConsoleHost.DefaultInitialSessionState = InitialSessionState.CreateDefault2();
-                        configuration = null;
-                    }
-                    else
-                    {
-                        ConsoleHost.DefaultInitialSessionState = InitialSessionState.CreateDefault2();
-                        configuration = null;
-                    }
-#else
+            if (args.Length > 0 && !String.IsNullOrEmpty(args[0]) && args[0].Equals("-isswait", StringComparison.OrdinalIgnoreCase))
+            {
+                Console.WriteLine("Attach the debugger to continue...");
+                while (!System.Diagnostics.Debugger.IsAttached) {
+                    Thread.Sleep(100);
+                }
+                System.Diagnostics.Debugger.Break();
                 ConsoleHost.DefaultInitialSessionState = InitialSessionState.CreateDefault2();
-                configuration = null;
-#endif
             }
             else
             {
-                //TODO : Deprecate RunspaceConfiguration and use InitialSessionState
-                configuration =
-                    RunspaceConfigForSingleShell.Create(consoleFilePath, out warning);
+                ConsoleHost.DefaultInitialSessionState = InitialSessionState.CreateDefault2();
             }
+#else
+            ConsoleHost.DefaultInitialSessionState = InitialSessionState.CreateDefault2();
+#endif
```

Comment:
How about change it to the following to remove 2 duplicate lines of `InitialSessionState.CreateDefault2()`?
```
#if DEBUG
            if (args.Length > 0 && !String.IsNullOrEmpty(args[0]) && args[0].Equals("-isswait", StringComparison.OrdinalIgnoreCase))
            {
                Console.WriteLine("Attach the debugger to continue...");
                while (!System.Diagnostics.Debugger.IsAttached) {
                    Thread.Sleep(100);
                }
                System.Diagnostics.Debugger.Break();
            }
#endif
            ConsoleHost.DefaultInitialSessionState = InitialSessionState.CreateDefault2();
```

Reply:
sounds good